### PR TITLE
Exit if sds socket not found

### PR DESCRIPTION
**Pull request by asayah**
**Please provide a description of this PR:**

fix: https://github.com/istio/istio/issues/45534

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ x] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "istio build-tools",
-  "image": "gcr.io/istio-testing/build-tools:master-daf61d16687130dc3cb60896f395f0d2df702669",
+  "image": "gcr.io/istio-testing/build-tools:master-a88efdd8afa93aad9a256a931ef89ead1baa14f9",
```
Comment (hzxuzhonghu) [inline .devcontainer/devcontainer.json:3]: ? can you rebase

**Review by bleggett — CHANGES_REQUESTED** [review]
It's not clear to me (based on this code or from the discussion in https://github.com/istio/istio/issues/45534) why we need a new option here?

If you define a custom socket path, and it isn't there, we should throw an error and not continue in _all cases_, shouldn't we?

File: pilot/cmd/pilot-agent/options/agent.go [hunk 2]
```diff
@@ -66,6 +66,7 @@ func NewAgentOptions(proxy *model.Proxy, cfg *meshconfig.ProxyConfig) *istioagen
 		ProxyDomain:                 proxy.DNSDomain,
 		IstiodSAN:                   istiodSAN.Get(),
 		DualStack:                   features.EnableDualStack,
+		ExitIfSDSSocketNotFound:     exitIfSDSSocketNotFoundEnv,
```
Comment (bleggett) [inline pilot/cmd/pilot-agent/options/agent.go:69]: Based on context in https://github.com/istio/istio/issues/45534#issuecomment-1631201311 - can I propose we just change this flag to `UseExternalWorkloadSDS`?

If it's set, we check for the alt socket and error if it's not there. If it isn't we don't.

File: releasenotes/notes/exit-if-sds-socket-not-found.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: security
+issue:
+- 45534
+releaseNotes:
+- |
+  **Added** prevent the istio-proxy from starting if the workload SDS socket is not found.
```
Comment (bleggett) [inline releasenotes/notes/exit-if-sds-socket-not-found.yaml:8]: Can we mention that

1. You need to set this variable to use external SDS now
2. what the variable is

**Review by bleggett — APPROVED** [review]
LGTM - ideally I'd prefer to actually not check/use the external socket at all if `UseExternalWorkloadSDS` is set, since I don't actually think there's a very strong case for maintaining back compat here, but this is better than nothing.

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (asayah) [reply .devcontainer/devcontainer.json]: done

File: releasenotes/notes/exit-if-sds-socket-not-found.yaml (hunk 3 above)
Comment (asayah) [reply releasenotes/notes/exit-if-sds-socket-not-found.yaml]: done

**Review by stevenctl — APPROVED** [review]

**Pull request merged**